### PR TITLE
Improves processes and their logging and downloads of URLs from VirtualFile

### PR DESCRIPTION
**Pull request by andyHa**
This is a breaking change, as some APIs in VirtualFile have changed as a boolean has been replaced by an enum.

Changes:
* Processes which encountered a warning are now marked and will be "filterable"
* ProcessLogs with a certain message type can now be limited
* These types and limits can now be supplied via an ExceptionHint
* i18n'd ProcessLog entries now still contain a translated message (in the current language) so that they're at least somehow "searchable"
* VirtualFile.loadFromUrl and VirtualFile.resolveOrLoadChildFromURL no support a mode to better control if and how data is fetched.

Fixes: OX-7270, OX-7332

### CODE REVIEW

File: src/main/resources/biz_de.properties [hunk 1]
```diff
@@ -645,6 +646,7 @@ ProcessController.sortDesc = Neueste zuerst
 ProcessController.unknownAction = Unbekannte Aktion: ${action}
 ProcessController.unknownOutput = Unbekannte Ausgabe: ${output}
 Processes.restarted = Der Prozess wurde erneut gestartet: ${reason}
+Processes.messageLimitReached = Die Nachricht '${type}' trat insgesamt ${count} mal auf, für eine bessere übersichtlichkeit wurden aber nur die ersten ${limit} Vorkommen aufgezeichnet.
```
Comment (ymo-sci) [inline src/main/resources/biz_de.properties:649]:
```suggestion
Processes.messageLimitReached = Die Nachricht '${type}' trat insgesamt ${count} Mal auf. Für bessere Übersichtlichkeit wurden aber nur die ersten ${limit} Vorkommen aufgezeichnet.
```

File: src/main/java/sirius/biz/process/ProcessEnvironment.java [hunk 2]
```diff
@@ -194,9 +198,27 @@ public void updateTitle(String newTitle) {
...
     @Override
     public void log(ProcessLog logEntry) {
+        if (Strings.isFilled(logEntry.getMessageType()) && logEntry.getMaxMessagesToLog() > 0) {
+            AtomicInteger messagesSoFar =
+                    messageCountsPerType.computeIfAbsent(logEntry.getMessageType(), this::countMessagesForType);
+            messagesSoFar.incrementAndGet();
+            if (messagesSoFar.incrementAndGet() >= logEntry.getMaxMessagesToLog()) {
+                return;
+            }
+            if (messagesSoFar.incrementAndGet() > logEntry.getMaxMessagesToLog()) {
```
Comment (idlira) [inline src/main/java/sirius/biz/process/ProcessEnvironment.java:208]: why 2x ifs (>= and >)?

File: src/main/java/sirius/biz/storage/layer3/VirtualFile.java [hunk 3]
```diff
@@ -1306,145 +1346,186 @@ public boolean loadFromUrl(URL url, boolean force) throws IOException {
...
-    private Tuple<Optional<String>, Optional<LocalDateTime>> requestHeaders(URL url) throws IOException {
-        Outcall headRequest = new Outcall(url);
-        headRequest.markAsHeadRequest();
-        Optional<String> contentDispositionPath = headRequest.parseFileNameFromContentDisposition();
-        Optional<LocalDateTime> lastModifiedHeader =
-                headRequest.getHeaderFieldDate(HttpHeaderNames.LAST_MODIFIED.toString());
+    private String parsePathFromUrl(URL url, Predicate<String> fileExtensionVerifier) {
+        // If the URL has a querystring, we check every parameter and determine if there is one with a valid
+        // filename. Otherwise, we use the filename as provided by the URL path itself...
+        QueryStringDecoder qsd = new QueryStringDecoder(url.toString(), StandardCharsets.UTF_8);
```
Comment (sabieber) [inline src/main/java/sirius/biz/storage/layer3/VirtualFile.java:1385]: Optional: Rename to `decoder`

**Review by sabieber — APPROVED** [review]

**Review by idlira — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/process/ProcessEnvironment.java [hunk 4]
```diff
@@ -194,9 +198,27 @@ public void updateTitle(String newTitle) {
...
+                return;
+            }
```
Comment (fhaScireum) [inline src/main/java/sirius/biz/process/ProcessEnvironment.java:210]: Does this code block make sense? Shouldn't it just be the bottom if-block?

**Review by fhaScireum — CHANGES_REQUESTED** [review]

**Review by meg1502 — APPROVED** [review]

File: src/main/java/sirius/biz/process/ProcessEnvironment.java (hunk 4 above)
Comment (andyHa) [reply src/main/java/sirius/biz/process/ProcessEnvironment.java]: +1

**Review by sabieber — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by jmuscireum — APPROVED** [review]

**Pull request merged**